### PR TITLE
✨ Source gong : add incremental Feature

### PR DESCRIPTION
**Pull request by KimPlv**
## What
Add Incremental feature

## How
- [Calls](https://gong.app.gong.io/settings/api/documentation#get-/v2/calls): use fromDateTime in String Parameter

- [answeredScorecards](https://gong.app.gong.io/settings/api/documentation#post-/v2/stats/activity/scorecards): use Filter with reviewFromDate in Request Body 
incremental custom implemented for this : {'filter': {'reviewFromDate': 'YYYY-MM-DD'}}

## Pre-merge Actions
*Expand the relevant checklist and delete the others.*

<details><summary><strong>Update Connector</strong></summary>

### Community member or Airbyter

- Connector version is set to `0.1.2`
- Documentation updated
    - `docs/integrations/source/gong.md` including changelog with an entry for the initial version. 
    - `docs/integrations/README.md`

</details>

----

| <a href="https://ellipsis.dev" target="_blank">Ellipsis</a> | :rocket: This PR description was created by [Ellipsis](https://www.ellipsis.dev) for commit b42def23c6bb38d37edd22b309ccf84bdf6e21ac.  | 
|--------|--------|

### Summary:
This PR adds incremental sync support to the `calls` and `answeredScorecards` streams in the Gong source connector, updates the connector version, and includes the necessary changes in the configuration files, unit tests, and documentation.

**Key points**:
- Added incremental sync support to the `calls` and `answeredScorecards` streams in the Gong source connector.
- Added a new class `IncrementalSingleBodyFilterCursor` in `components.py` to handle the incremental sync.
- Updated `metadata.yaml` and `configured_catalog.json` files to reflect the new version and the added incremental sync support.
- Updated `spec.yaml` to include the `start_date` in the connection specification.
- Added a new unit test to test the request filter options.
- Updated the documentation to reflect the new changes.

----
Generated with :heart: by [ellipsis.dev](https://www.ellipsis.dev)

### CODE REVIEW

File: airbyte-integrations/connectors/source-gong/source_gong/spec.yaml [hunk 1]
```diff
@@ -22,8 +23,7 @@ connectionSpecification:
       type: string
       title: Start date
       pattern: ^[0-9]{4}-[0-9]{2}-[0-9]{2}T[0-9]{2}:[0-9]{2}:[0-9]{2}Z$
+      format: date-time
       description: >-
         The date from which to list calls, in the ISO-8601 format; if not specified, the calls start with the earliest recorded call.
         For web-conference calls recorded by Gong, the date denotes its scheduled time, otherwise, it denotes its actual start time.
-      examples:
-        - "2018-02-18T08:00:00Z"
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-gong/source_gong/spec.yaml:29]: Any reason to remove example?

File: airbyte-integrations/connectors/source-gong/source_gong/components.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        return self._get_request_filter_options(RequestOptionType.body_json, stream_slice)
+
+    def _get_request_filter_options(self, option_type: RequestOptionType, stream_slice: Optional[StreamSlice]) -> Mapping[str, Any]:
+        options: MutableMapping[str, Any] = {}
+        if not stream_slice:
+            return options
+        if self.start_time_option and self.start_time_option.inject_into == option_type:
+            field_name, sub_field_name = self.start_time_option.field_name.eval(config=self.config).replace(" ", "").split(",")
+            options[field_name] = {sub_field_name: stream_slice.get(self._partition_field_start.eval(self.config))}
+        return options
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-gong/source_gong/components.py:26]: Can you add unit test for this component?

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
I left some comments. Thanks for the contribution!

**Comment by marcosmarxm** [comment]
@ellipsis-dev review this contribution.

File: airbyte-integrations/connectors/source-gong/source_gong/spec.yaml [hunk 3]
```diff
@@ -6,6 +6,7 @@ connectionSpecification:
   required:
     - access_key
     - access_key_secret
+    - start_date
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-gong/source_gong/spec.yaml:9]: Making this as required will introduce a breaking change to existing syncs.

File: airbyte-integrations/connectors/source-gong/source_gong/manifest.yaml [hunk 4]
```diff
@@ -80,15 +104,27 @@ definitions:
...
+    incremental_sync:
+      type: CustomIncrementalSync
+      class_name: source_gong.components.IncrementalSingleBodyFilterCursor
+      cursor_field: reviewTime
+      cursor_datetime_formats:
+        - "%Y-%m-%dT%H:%M:%S.%f%z"
+      datetime_format: "%Y-%m-%d"
+      start_datetime:
+        type: MinMaxDatetime
+        datetime: "{{ config['start_date'] }}"
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-gong/source_gong/manifest.yaml:116]:
```suggestion
        datetime: "{{ config,get('start_date', now().subtract(days=-180) }}"
```
What about adding a default value when users didn't specify the start date? This way it won't introduce a breaking change.

**Review by marcosmarxm — CHANGES_REQUESTED** [review]

**Comment by marcosmarxm** [comment]
@CodiumAI-Agent /review

File: airbyte-integrations/connectors/source-gong/source_gong/manifest.yaml (hunk 4 above)
Comment (KimPlv) [reply airbyte-integrations/connectors/source-gong/source_gong/manifest.yaml]: you're right, added :white_check_mark:

**Comment by FilahAnas** [comment]
Hi @marcosmarxm, I noticed that this PR has been closed. Is there anything missing on our end? We are interested in adding the Incremental feature to this connector. Please let me know if there is anything we should correct or address. Thank you! :D

**Comment by marcosmarxm** [comment]
Sorry about that @KimPlv and @FilahAnas I totally lost track of this contribution. Do you mind submit again I'll take priority to get it done.

**Comment by FilahAnas** [comment]
> Sorry about that @KimPlv and @FilahAnas I totally lost track of this contribution. Do you mind submit again I'll take priority to get it done.

No problem, I ll submit again. Thank you !

**Comment by cbuckle1** [comment]
@KimPlv - any reason not to add extensivecalls as incremental?

**Pull request closed without merging**